tests/mock_network/accumulate.rs

```diff
@@ -56,27 +51,6 @@ fn messages_accumulate_with_quorum() {
     let _ = poll_all(&mut nodes, &mut []);
     expect_no_event!(nodes[0]);
 
-    // If there are `quorum` senders but they all only sent hashes, nothing can accumulate.
-    // Only after one of the first three nodes, which are closest to `src.name()`, has sent the full
-    // message, it accumulates.
-    let message_id = MessageId::new();
-    for node in nodes
-        .iter_mut()
-        .skip(dg_size)
-        .take(min(quorum, section_size - dg_size))
-    {
-        send(node, &dst, message_id);
```

Comment:
Would like to confirm is this block of testing code no longer related?
According to the comment, it seems the test shall still be applied?

And same question for the below removal as well.

Reply:
It became irrelevant even earlier, actually, it just worked because of unrelated reasons.

In earlier versions of the code, only the accumulating nodes would generate the full message, and the others would only send signed hashes. Because of that, a message couldn't accumulate until an accumulating node also generated a signature and the message itself along with it. That's the case this code was testing.

However, since RMD and SMD, all nodes send full messages when they send signatures. This means that the case this was testing is now impossible - whichever nodes send the signatures, the full message is available and it should accumulate correctly. So these pieces of code are now irrelevant.

Prior to this removal what was happening was that if no accumulating node signed the message (accumulating nodes being the ones that were a part of the delivery group), there wasn't enough of the other nodes for the message to accumulate, as there was N/3 accumulating nodes, and we required over 2N/3 signatures for a message to accumulate. This wasn't really what the code intended to test, but the test still worked.

This now changes, though - N/3 signatures are enough, which means that other nodes' signatures are enough and this code makes no sense whatsoever. That's why it is being removed.

I hope that cleared things up ;)

Reply:
Ah, thank you very much for the explanation.  :)

Reply:
Great explanation, could you put it in PR description or the commit @fizyk20 ?